packages/scaffold/src/client.ts

```diff
@@ -43,11 +49,51 @@ export class Web3ModalScaffold {
     ModalController.open(options)
   }
 
+  public unsubscribe: (() => void)[] = []
+
   public async close() {
     await this.initOrContinue()
     ModalController.close()
   }
 
+  public getThemeMode() {
+    return ThemeController.state.themeMode
+  }
+
+  public getThemeVariables() {
+    return ThemeController.state.themeVariables
+  }
+
+  public setThemeMode(themeMode: ThemeMode) {
+    ThemeController.setThemeMode(themeMode)
+  }
+
+  public setThemeVariables(themeVariables: ThemeVariables) {
+    ThemeController.setThemeVariables(themeVariables)
+  }
+
+  public subscribeThemeMode(setter: (value: ThemeMode) => void) {
+    this.unsubscribe.push(
+      ThemeController.subscribeKey('themeMode', val => {
```

Comment:
The unsubscribe array pattern is needed only for our web-components. Here, let's remove it + that unbount function. Instead just `return ThemeController.subscribeKey...` this returns unsubscribe method that we can use latter on i.e.

```
const unsubscribe = modal.subscribeTheme()
unsubscribe()
```

You can use this unsubscribe method in a hook's cleanup for example.

Also, let's combine theme subscriber into one function i.e. `subscribeTheme` that subscribes to whole ThemeController state.